app/controllers/test_sets_controller.rb

```diff
@@ -2,11 +2,12 @@ class TestSetsController < ApplicationController
   allow_unauthenticated_access
 
   def index
-    @test_sets = TestSet.includes(:tasks).published
+    @test_sets = policy_scope(TestSet)
```

Comment:
```suggestion
    @test_sets = policy_scope(TestSet).includes(:tasks).published
```

Also, I'm considering adding `published` to the policy scope, but once it is done and the admin panel will be using policies, a new one (dedicated to the admin) needs to be created.